src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -7407,6 +7407,28 @@ HRESULT DacDbiInterfaceImpl::GetReJitInfo(VMPTR_MethodDesc vmMethod, CORDB_ADDRE
     return S_OK;
 }
 
+HRESULT DacDbiInterfaceImpl::AreOptimizationsDisabled(VMPTR_Module vmModule, mdMethodDef methodTk, OUT BOOL* pOptimizationsDisabled)
+{
+    DD_ENTER_MAY_THROW;
+#ifdef FEATURE_REJIT
+    PTR_Module pModule = vmModule.GetDacPtr();
+    if (pModule == NULL || pOptimizationsDisabled == NULL || TypeFromToken(methodTk) != mdtMethodDef)
+    {
+        return E_INVALIDARG;
+    }
+    {
+        CodeVersionManager::LockHolder codeVersioningLockHolder;
+        CodeVersionManager * pCodeVersionManager = pModule->GetCodeVersionManager();
+        ILCodeVersion activeILVersion = pCodeVersionManager->GetActiveILCodeVersion(pModule, methodTk);
+        *pOptimizationsDisabled = activeILVersion.IsDeoptimized();
+    }
```

Comment:
```suggestion
    CodeVersionManager * pCodeVersionManager = pModule->GetCodeVersionManager();
    ILCodeVersion activeILVersion = pCodeVersionManager->GetActiveILCodeVersion(pModule, methodTk);
    *pOptimizationsDisabled = activeILVersion.IsDeoptimized();
```

You don't need `codeVersioningLockHolder` or the scope here.

Reply:
You need to hold the lock to call `GetActiveILCodeVersion`

Reply:
Not in the DAC

Reply:
I don't know what mechanism is in place though to ensure the DAC doesn't pick a code version while a BG jit happens in the target process.